april_tag_updates/new_april.py

```diff
@@ -0,0 +1,151 @@
+            cv2.line(image, topRightPoint, bottomRightPoint, (0, 255, 0), 2)
+            cv2.line(image, bottomRightPoint, bottomLeftPoint, (0, 255, 0), 2)
+            cv2.line(image, bottomLeftPoint, topLeftPoint, (0, 255, 0), 2)
+            cX = int((topLeft[0] + bottomRight[0])//2)
+            cY = int((topLeft[1] + bottomRight[1])//2)
+            measure = abs(5/(topLeft[0]-cX))
+            cv2.circle(image, (cX, cY), 4, (255, 0, 0), -1)
+            cv2.putText(image, str(
+                int(markerId)), (int(topLeft[0]-10), int(topLeft[1]-10)), cv2.FONT_HERSHEY_COMPLEX, 1, (0, 0, 255))
+            Dist.append((cX, cY))
+            # print(arucoDict)
+            if len(Dist) == 0:
+                if Line_Pts is not None:
+                    Dist = Line_Pts
+            if len(Dist) == 2:
+                Line_Pts = Dist
+            if len(Dist) == 2:
+                cv2.line(image, Dist[0], Dist[1], (255, 0, 255), 2)
+                ed = ((Dist[0][0] - Dist[1][0])**2 +
+                ((Dist[0][1] - Dist[1][1])**2))**(0.5)
+                cv2.putText(image, str(int(measure*(ed))) + "cm", (int(300), int(
+                300)), cv2.FONT_HERSHEY_COMPLEX, 1, (0, 0, 255))
+    cv2.imshow(window, image)
+
+    Dist2 = []
+    ret2, image2 = cap2.read()
+    image2 = imutils.resize(image2, width=800)
+    (corners2, ids2, rejected2) = cv2.aruco.detectMarkers(
+        image2, arucoDict, parameters=arucoParams)
+    if len(corners2) <= 0:
+        if CACHED_PTS2 is not None:
+            corners2 = CACHED_PTS2
+    if len(corners2) > 0:
+        CACHED_PTS2 = corners2
+        if ids2 is not None:
+           ids2 = ids2.flatten()
+           CACHED_IDS2 = ids2
+        else:
+           if CACHED_IDS2 is not None:
+              ids2 = CACHED_IDS2
+        if len(corners2) < 2:
+           if len(CACHED_PTS2) >= 2:
+              corners2 = CACHED_PTS2
+        for (markerCorner, markerId) in zip(corners2, ids2):
+            print("[INFO] Marker detected2")
+            corners_abcd = markerCorner.reshape((4, 2))
+            (topLeft, topRight, bottomRight, bottomLeft) = corners_abcd
+            topRightPoint = (int(topRight[0]), int(topRight[1]))
+            topLeftPoint = (int(topLeft[0]), int(topLeft[1]))
+            bottomRightPoint = (int(bottomRight[0]), int(bottomRight[1]))
+            bottomLeftPoint = (int(bottomLeft[0]), int(bottomLeft[1]))
+            cv2.line(image2, topLeftPoint, topRightPoint, (0, 255, 0), 2)
+            cv2.line(image2, topRightPoint, bottomRightPoint, (0, 255, 0), 2)
+            cv2.line(image2, bottomRightPoint, bottomLeftPoint, (0, 255, 0), 2)
+            cv2.line(image2, bottomLeftPoint, topLeftPoint, (0, 255, 0), 2)
+            cX = int((topLeft[0] + bottomRight[0])//2)
+            cY = int((topLeft[1] + bottomRight[1])//2)
+            measure2 = abs(5/(topLeft[0]-cX))
+            cv2.circle(image2, (cX, cY), 4, (255, 0, 0), -1)
+            cv2.putText(image2, str(

```

Comment:
the logic for image 2 looks exceedingly similar to image 1. Can we instead make a function and call it on image 1 and image2?